website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,540 @@
+---
+title: Forms
+---
+
+In the previous sections we learned how to start building the actual User Interface by using UI components and fetching data. One of the most common thing to implement in a User Interface is forms.
+
+To facilitate the development of UI forms, we provide many different ready-to-use input and field components based on [commercetools Design System](https://github.com/commercetools/ui-kit).
+
+# Form state management
+
+Forms generally consist of a group of inputs that users interact with and eventually some actions will be triggered.
+All the user interactions need to be tracked, managed, and reflected in the UI. This is where the form needs to have some state to be able to handle all of that.
+
+At commercetools we have settled on using [Formik](https://formik.org) as the preferred form state management library when building User Interfaces.
+
+Formik comes with several built-in features such as validations, array fields, input states, async submission, etc.
+
+At a bare minimum, implementing a form could look something like this:
+
+```jsx
+import { useFormik } from 'formik';
+import TextField from '@commercetools-uikit/text-field';
+import PrimaryButton from '@commercetools-uikit/primary-button';
+import validate from './validate';
+
+const ChannelsForm = () => {
+  const formik = useFormik({
+    initialValues: {
+      // A Channel's `key`: https://docs.commercetools.com/api/projects/channels
+      key: '',
+    },
+    validate,
+    onSubmit: async (formikValues) => {
+      alert(`key: ${formikValues.key}`);
+      // Do something async
+    },
+  });
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
+      <TextField
```

Comment:
Should we recommend using `useField` to spread in the props easier?

Reply:
Yeah, we could have to define the a `FormKeyField` or something as a child to be able to do that.